crud/src/main/java/com/redhat/lightblue/mediator/Mediator.java

```diff
@@ -418,4 +423,28 @@ private void updatePredefinedFields(OperationContext ctx,CRUDController controll
             controller.updatePredefinedFields(ctx,doc);
         }
     }
+
+    /**
+     * Checks if the caller has access to all the query fields. Returns false if not, and sets the error status in ctx
+     */
+    private boolean checkQueryAccess(OperationContext ctx,QueryExpression query) {
+        if(query!=null) {
+            CompositeMetadata md=ctx.getTopLevelEntityMetadata();
+            FieldAccessRoleEvaluator eval=new FieldAccessRoleEvaluator(md,ctx.getCallerRoles());
+            List<FieldInfo> fields=query.getQueryFields();
+            LOGGER.debug("Checking access for query fields {}",fields);
+            for(FieldInfo field:fields) {
+                LOGGER.debug("Access checking field {}",field.getAbsFieldName());
+                if(eval.hasAccess(field.getAbsFieldName(),FieldAccessRoleEvaluator.Operation.find)) {
+                    LOGGER.debug("Field {} is readable",field.getAbsFieldName());
+                } else {
+                    LOGGER.debug("Field {} is not readable",field.getAbsFieldName());
+                    ctx.addError(Error.get(CrudConstants.ERR_NO_ACCESS, field.getAbsFieldName().toString()));
+                    ctx.setStatus(OperationStatus.ERROR);
+                    return false;
```

Comment:
Ready for review.

On Tue, Jan 13, 2015 at 11:45 AM, Burak Serdar <bserdar@ieee.org> wrote:

> ok, will add that.
>
> On Tue, Jan 13, 2015 at 11:15 AM, Naveen Malik <notifications@github.com>
> wrote:
>
>> In crud/src/main/java/com/redhat/lightblue/mediator/Mediator.java
>> <https://github.com/lightblue-platform/lightblue-core/pull/279#discussion-diff-22881628>
>> :
>>
>> > +     */
>> > +    private boolean checkQueryAccess(OperationContext ctx,QueryExpression query) {
>> > +        if(query!=null) {
>> > +            CompositeMetadata md=ctx.getTopLevelEntityMetadata();
>> > +            FieldAccessRoleEvaluator eval=new FieldAccessRoleEvaluator(md,ctx.getCallerRoles());
>> > +            List<FieldInfo> fields=query.getQueryFields();
>> > +            LOGGER.debug("Checking access for query fields {}",fields);
>> > +            for(FieldInfo field:fields) {
>> > +                LOGGER.debug("Access checking field {}",field.getAbsFieldName());
>> > +                if(eval.hasAccess(field.getAbsFieldName(),FieldAccessRoleEvaluator.Operation.find)) {
>> > +                    LOGGER.debug("Field {} is readable",field.getAbsFieldName());
>> > +                } else {
>> > +                    LOGGER.debug("Field {} is not readable",field.getAbsFieldName());
>> > +                    ctx.addError(Error.get(CrudConstants.ERR_NO_ACCESS, field.getAbsFieldName().toString()));
>> > +                    ctx.setStatus(OperationStatus.ERROR);
>> > +                    return false;
>>
>> Since we have all the fields, why not continue processing to find out if
>> there are additional things the caller is not authorized to access, add
>> each as an error?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/lightblue-platform/lightblue-core/pull/279/files#r22881628>
>> .
>>
>
>